src/generic/datavgen.cpp

```diff
@@ -6450,9 +6450,23 @@ void wxDataViewCtrl::EnsureVisibleRowCol( int row, int column )
     int last = m_clientArea->GetLastFullyVisibleRow();
     if( row < first )
         m_clientArea->ScrollTo( row, column );
-    else if( row > last )
-        m_clientArea->ScrollTo( row - last + first, column );
-    else
+    else if (row > last)
+    {
+        if (!HasFlag(wxDV_VARIABLE_LINE_HEIGHT))
+            m_clientArea->ScrollTo(row - last + first, column);
+        else
+        {
+            // calculate scroll position based on last visible item
+            const int itemStart = m_clientArea->GetLineStart(row);
+            const int itemHeight = m_clientArea->GetLineHeight(row);
+            const int clientHeight = m_clientArea->GetSize().y;
+            int scrollX, scrollY;
+            GetScrollPixelsPerUnit(&scrollX, &scrollY);
+            int scrollPosY = (itemStart + itemHeight - clientHeight) / scrollY;
+            Scroll(-1, scrollPosY + 1); // +1 due to integer division, always show whole item
+        }
+    }
+    else if (row == first)
```

Comment:
We don't need this test, perhaps just leave it as a comment but testing for it is IMO confusing as it seems to indicate that the test might be false -- which is never the case.
```suggestion
    else // row == first
```

Reply:
I try to explain. Currently the else case scolls always to first row even if row is somewhere between fist and last row. for variable line height this results in the following behaviour (gif):
![scroll_up](https://user-images.githubusercontent.com/6022600/211997510-d077fdae-7913-4f17-ad1f-5b2195cff543.gif)
Because of GetFirstVisibleRow returns the first not fully visible row and ScrollTo(newRow) makes newRow fully visible. Maybe that is already a problem. Indeed it is not necessary (for fixed and variable line height) to scroll if the row is already between the visible rows so i added this test.
But yes the code could look cleaner with:
`
if( row <= first )
    m_clientArea->ScrollTo( row, column );
else if (row > last) {...}
`